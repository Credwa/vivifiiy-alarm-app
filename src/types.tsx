import { AlarmInterface } from './interfaces';

export type RootStackParamList = {
  Root: undefined;
  NotFound: undefined;
};

export type BottomTabParamList = {
  Home: undefined;
  Alarms: undefined;
  Insights: undefined;
  Settings: undefined;
};

export type HomeTabParamList = {
  HomeTabScreen: undefined;
};

export type AlarmsTabParamList = {
  AlarmsTabScreen: undefined;
};

export type InsightsTabParamList = {
  InsightsTabScreen: undefined;
};

export type SettingsTabParamList = {
  SettingsTabScreen: undefined;
};

export type meridiem = 'AM' | 'PM';

export type twelveHrTime = { hour: string; minute: string; meridiem: meridiem };

export type alarmObject = { [x: string]: AlarmInterface };
