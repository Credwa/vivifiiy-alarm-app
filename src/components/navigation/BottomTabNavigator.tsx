import Colors from '@/constants/Colors';
import AlarmsScreen from '@/screens/AlarmsScreen';
import HomeScreen from '@/screens/HomeScreen';
import InsightsScreen from '@/screens/InsightsScreen';
import SettingsScreen from '@/screens/SettingsScreen';
import {
  AlarmsTabParamList,
  BottomTabParamList,
  HomeTabParamList,
  InsightsTabParamList,
  SettingsTabParamList
} from '@/types';
import { FontAwesome, Ionicons, SimpleLineIcons } from '@expo/vector-icons';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';
import React from 'react';

const BottomTab = createBottomTabNavigator<BottomTabParamList>();

export default function BottomTabNavigator() {
  return (
    <BottomTab.Navigator
      initialRouteName="Home"
      tabBarOptions={{
        activeTintColor: Colors.white,
        inactiveTintColor: Colors.greyLight2,
        tabStyle: { backgroundColor: Colors.blueMediumCardBody },
        style: { backgroundColor: Colors.blueMediumCardBody, borderTopColor: Colors.blueDark }
      }}
    >
      <BottomTab.Screen
        name="Home"
        component={HomeTabNavigator}
        options={{
          tabBarIcon: ({ color }) => <FontAwesome size={24} color={color} name="moon-o" />
        }}
      />
      <BottomTab.Screen
        name="Alarms"
        component={AlarmsTabNavigator}
        options={{
          tabBarIcon: ({ color }) => <Ionicons name="md-alarm" size={26} color={color} />
        }}
      />
      <BottomTab.Screen
        name="Insights"
        component={InsightsTabNavigator}
        options={{
          tabBarIcon: ({ color }) => <SimpleLineIcons name="bulb" size={24} color={color} />
        }}
      />
      <BottomTab.Screen
        name="Settings"
        component={SettingsTabNavigator}
        options={{
          tabBarIcon: ({ color }) => <SimpleLineIcons name="settings" size={24} color={color} />
        }}
      />
    </BottomTab.Navigator>
  );
}

const HomeTabStack = createStackNavigator<HomeTabParamList>();

function HomeTabNavigator() {
  return (
    <HomeTabStack.Navigator>
      <HomeTabStack.Screen name="HomeTabScreen" component={HomeScreen} options={{ headerShown: false }} />
    </HomeTabStack.Navigator>
  );
}

const AlarmsTabStack = createStackNavigator<AlarmsTabParamList>();

function AlarmsTabNavigator() {
  return (
    <AlarmsTabStack.Navigator>
      <AlarmsTabStack.Screen name="AlarmsTabScreen" component={AlarmsScreen} options={{ headerShown: false }} />
    </AlarmsTabStack.Navigator>
  );
}

const InsightsTabStack = createStackNavigator<InsightsTabParamList>();

function InsightsTabNavigator() {
  return (
    <InsightsTabStack.Navigator>
      <InsightsTabStack.Screen name="InsightsTabScreen" component={InsightsScreen} options={{ headerShown: false }} />
    </InsightsTabStack.Navigator>
  );
}

const SettingsTabStack = createStackNavigator<SettingsTabParamList>();

function SettingsTabNavigator() {
  return (
    <SettingsTabStack.Navigator>
      <SettingsTabStack.Screen name="SettingsTabScreen" component={SettingsScreen} options={{ headerShown: false }} />
    </SettingsTabStack.Navigator>
  );
}
