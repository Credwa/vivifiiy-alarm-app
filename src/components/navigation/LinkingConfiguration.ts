import * as Linking from 'expo-linking';

export default {
  prefixes: [Linking.makeUrl('/')],
  config: {
    screens: {
      Root: {
        screens: {
          Home: {
            screens: {
              HomeScreen: 'home'
            }
          },
          Alarms: {
            screens: {
              AlarmsScreen: 'alarms',
              NewAlarmScreen: 'newAlarm',
              EditAlarmScreen: 'editAlarm'
            }
          },
          Insights: {
            screens: {
              InsightsScreen: 'insights'
            }
          },
          Settings: {
            screens: {
              SettingsScreen: 'settings'
            }
          }
        }
      },
      NotFound: '*'
    }
  }
};
