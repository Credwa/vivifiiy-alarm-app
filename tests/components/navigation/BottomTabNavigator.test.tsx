import React from 'react';
import renderer from 'react-test-renderer';
import BottomTabNavigator from '@/components/navigation/BottomTabNavigator';
import { createStackNavigator } from '@react-navigation/stack';
import { RootStackParamList } from '@/types';
import { NavigationContainer } from '@react-navigation/native';
import LinkingConfiguration from '@/components/navigation/LinkingConfiguration';
jest.useFakeTimers();
describe('<BottomTabNavigator />', () => {
  it(`renders correctly`, () => {
    const Stack = createStackNavigator<RootStackParamList>();
    const tree = renderer
      .create(
        <NavigationContainer linking={LinkingConfiguration}>
          <Stack.Navigator screenOptions={{ headerShown: false }}>
            <Stack.Screen name="Root" component={BottomTabNavigator} />
          </Stack.Navigator>
        </NavigationContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
