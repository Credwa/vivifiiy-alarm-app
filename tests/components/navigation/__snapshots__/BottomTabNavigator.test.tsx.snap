// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomTabNavigator /> renders correctly 1`] = `null`;
